pkg/buildlog/buildlog.go

```diff
@@ -0,0 +1,45 @@
+package buildlog
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+)
+
+// Entry represents a single line in a Melange build log.
+type Entry struct {
+	Arch, Origin, Package, FullVersion string
+}
+
+// Parse parses a Melange build log into a slice of entries.
+func Parse(r io.Reader) ([]Entry, error) {
+	splitFunc := func(ru rune) bool {
+		return string(ru) == "|"
+	}
+
+	var entries []Entry
+
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		fields := strings.FieldsFunc(line, splitFunc)
```

Comment:
Why this instead of `strings.Fields(line, "|")` ?

Reply:
Maybe I missed it — is that a thing? https://pkg.go.dev/strings#Fields

Reply:
I'll resolve for now but I can follow up with a PR if there's a better way here!